internal/status/status.go

```diff
@@ -143,16 +145,20 @@ func (s *Status) Details() []interface{} {
 
 // Error is an alias of a status proto. It implements error and Status,
 // and a nil Error should never be returned by this package.
-type Error spb.Status
+type Error struct {
+	e spb.Status
```

Comment:
I think we could simply store a pointer to `spb.Status` here for a couple of reasons. When `Error` is created in `Status.Err`, the proto wrapped in `Status` is cloned before passing it on to `Error`. Similarly, in `Error.GRPCStatus`, the proto is cloned. So, we can avoid another copy by storing a pointer here.